### DISCUSSION TITLE
Are real time calculated Metrics needed?

### DISCUSSION THREAD

**Discussion by schnuerle** [openmobilityfoundation/curb-data-specification#30] [Data Specification]
What are the use cases for real time Metrics calculations for [consumers](https://github.com/openmobilityfoundation/curb-data-specification/discussions/29) of the data?  Is a city, agency, or commercial entity going to need real time Metrics? We think not.

There is already real time availability optionally included in Curb Spaces and Zones.

Can Metrics be historic, and calculated, say, at the end of every month and go back in time after that? Does anyone need metrics sooner than monthly (eg, weekly, daily)?

**Comment by jacobmalleau** [comment]
Agreed that metrics don't need to be real time. 
Can't there be an option to have access to metrics sooner than monthly? I don't see why not. I could see city agencies and mobility companies wanting to review metrics while they are still relevant (an event or construction that just happened, etc.).

**Comment by marisamangan** [comment]
If Metrics are determined to remain historic, can it be set up so a city can pull the last week (or combination of weeks) from a specified date? This may allow for some custom historical data pulls without needing to be real-time.

**Comment by schnuerle** [comment]
Maybe if we use this flat file option https://github.com/openmobilityfoundation/curb-data-specification/discussions/31 where counts of each metric are returned by the hour, we could leave it up to the agencies and say "Agencies may return data as soon as desired or when its available after metrics are calculated, e.g., after a month, week, day, or hour."

**Comment by bhamlinSDOT** [comment]
Also agree that I don't think Metrics need to be real-time. I honestly can't think of a reason where we would need that information.

I also think leaving it up to the agency makes a lot of sense. It allows flexibility for agencies that might not have the capacity to provide quick Metric releases to either a flat file or dynamic endpoint.

**Reply by tomcarr178** [reply]
Agree with Brian here. If we eventually see a need for real-time, then a future update of CDS can address it.

**Comment by schnuerle** [comment]
Agreed that we won't be doing Metrics in real-time. Thought the current wording says that the update frequency is up to the agency:

https://github.com/openmobilityfoundation/curb-data-specification/tree/feature-release-work-1/metrics#update-frequency

"At least monthly is recommended but it may be longer, or weekly, daily, hourly, or even more frequently."
